qemu/tests/vsock_negative_test.py

```diff
@@ -0,0 +1,73 @@
+import random
+import logging
+import aexpect
+
+from avocado.utils import process
+
+from virttest import utils_misc
+from virttest import error_context
+
+from . import vsock_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Vsock negative test
+
+    S1):
+    1. Boot guest with vhost-vsock-pci device
+    2. Download and compile nc-vsock on both guest and host
+    3. Connect guest CID(on host) without listening port inside guest
+    S2):
+    1. Boot guest with vhost-vsock-pci device
+    2. Download and compile nc-vsock on both guest and host
+    3. Send data from guest, nc-vsock -l $port < tmp_file
+    4. Receive data from host, nc-vsock $guest_cid $port
+    5. Interrupt nc-vsock process during transfering data on host
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    nc_vsock_bin = vsock_test.compile_nc_vsock(test, vm, session)
+    port = random.randrange(1, 6000)
+    vsock_dev = params["vsocks"].split()[0]
+    guest_cid = vm.devices.get(vsock_dev).get_param("guest-cid")
+    nc_vsock_cmd = "%s %s %s" % (nc_vsock_bin, guest_cid, port)
+    connected_str = "Connection reset by peer"
+    error_context.context("Connect vsock from host without"
+                          " listening on guest.")
+    try:
+        process.system_output(nc_vsock_cmd)
+    except process.CmdError as e:
+        if connected_str not in e.result:
+            test.fail("The connection does not failed as expected.")
+    else:
+        test.fail("The connection success, while it is expected to fail.")
+    finally:
+        session.close()
+
+    session = vm.wait_for_login()
+    tmp_file = "/tmp/vsock_file_%s" % utils_misc.generate_random_string(6)
+    cmd_generate = "dd if=/dev/urandom of=%s count=10000 bs=1M" % tmp_file
+    session.cmd_status(cmd_generate)
+    cmd_transfer = "%s -l %s < %s &" % (nc_vsock_bin, port, tmp_file)
+    error_context.context("Transfer file from guest via command: %s"
+                          % cmd_transfer, logging.info)
+    session.cmd_status(cmd_transfer)
+    cmd_receive = "%s %s %s > %s" % (nc_vsock_bin, guest_cid, port, tmp_file)
+    rec_session = aexpect.Expect(cmd_receive,
+                                 auto_close=True,
+                                 output_func=utils_misc.log_line,
+                                 output_params=("%s.log" % tmp_file,))
```

Comment:
Is there many cases need transfer data between host and guest through nc-vsock ?  we can make it a module, such as vsock_test.compile_nc_vsock,  introduce a basic class is ok too.

Reply:
@yanan-fu No more so far, and it will be 1 line or 2 lines for each function(send/receive) if package it to functions, I think that's meaningless.

Reply:
@yanan-fu Or we might package the whole test from L#54 - L#66 to a function called `transfer_data_from_guest_to_host_test`, and return guest session together with host session, what do you think?

Reply:
@vivianQizhu  Yes,  that is what i mean. Normally, guest send and host receive always be used together.  can make them together as a module and reuse them.  
For the return value,  host session is ok,  and i am not very sure about the guest session now, as it is a not a real nc-vsock session,  i mean not a session of `cmd_transfer` (nc-vsock  -l $port < $file &).